### ISSUE TITLE
Expose new Analytics All Traffic widget in production

### ISSUE THREAD

**Issue by felixarntz** [google/site-kit-wp#2429]
Once ready the new `DashboardAllTrafficWidgetV2` widget should replace the `DashboardAllTrafficWidget` widget and be exposed in production.

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* The `DashboardAllTrafficWidget` component and any sub-components that are only used by that one should be removed. All references to that widget should be removed too, including tests, Storybook etc.
* The `DashboardAllTrafficWidgetV2` should be renamed to `DashboardAllTrafficWidget`, all references should be updated accordingly. Its widget registration should be updated to drop the v2 indicator from the widget slug as well.
* The heading of both `*AllTraffic` widget areas should be updated to: _Your Traffic at a Glance_
* Pending on whether the widget API is not live yet, we'll need to also replace the legacy All Traffic widget with the new component (outside of the widget API) and updated the area heading and description.

## Implementation Brief

**Widgets**

* Remove `DashboardAllTrafficWidget` component
* Remove `AcquisitionSources` component
* Remove `AcquisitionPieChart` component
* Rename `DashboardAllTrafficWidgetV2` to `DashboardAllTrafficWidget`, change `<Widget>`'s `slug` to `analyticsAllTraffic`  and update all the references accordingly.
* Remove `analyticsAllTrafficV2` widget registration in `assets/js/modules/analytics/index.js`. Update `analyticsAllTraffic` registration to use the new `DashboardAllTrafficWidget`

**Stories**

* Remove `Analytics Module/Components/Page Dashboard/All Traffic Widget`
* Rename `Analytics Module/Components/Dashboard/All Traffic Widget V2` to `Analytics Module/Components/Dashboard/All Traffic Widget` and update the `Component` to use the new `DashboardAllTrafficWidget`
* Update `Analytics Module/Components/Page Dashboard/All Traffic Widget` to use the newer v2 of `DashboardAllTrafficWidget`

**Language Strings**

* Update `title` for `AREA_DASHBOARD_ALL_TRAFFIC` and `AREA_PAGE_DASHBOARD_ALL_TRAFFIC` to `Your Traffic at a Glance` in `assets/js/googlesitekit/widgets/register-defaults.js`

For legacy support:
* Add a `LegacyAllTraffic` component in `assets/js/modules/analytics/components/dashboard/DashboardAllTrafficWidget`.
* Using the code from `assets/js/modules/pagespeed-insights/components/dashboard/LegacyDashboardSpeed.js` as reference, create a wrapper around the new `DashboardAllTrafficWidget`. 
 	* Omit the conditionals around `description`, it will be used as a hard coded value.
	* The `title`  will be `Your Traffic at a Glance` and the `description` will be `How people found your site.`
* Add a filter in `assets/js/modules/analytics/index.legacy.js` to display the `DashboardAllTraffic` component.
	```js
	const legacyDashboardAllTraffic = createAddToFilter( <LegacyDashboardAllTraffic /> );
	```
	
	```js
	addFilter( 'googlesitekit.DashboardModule',  
	 'googlesitekit.Analytics',  
	 legacyDashboardAllTraffic, 8 );
	```

### Test Coverage

* No new tests to be added

### Visual Regression Changes

* The references to `All Traffic Widget V2` must be removed and references to `All Traffic Widget` be replaced with the newer v2 references.

## QA Brief

* `All Traffic Widget` story under `Dashboard` and `Page Dashboard` must be replaced with the v2 of the widget.
* `All Traffic Widget V2` story under `Dashboard` must be removed.
* On the dashboard, the All Traffic Widget must be updated to the v2 and `All Traffic` heading must be changed to `Your Traffic at a Glance`

## Changelog entry

* Launch new All Traffic widget in the Site Kit dashboard, which surfaces more detailed information about channels, includes a time-based graph, and surfaces similar metrics for countries and devices the site is accessed from.

**Comment by aaemnnosttv** [comment]
@kostyalmm would you please add an estimate?

**Comment by felixarntz** [comment]
@kostyalmm The IB looks good in terms of exposing the new All Traffic widget as a replacement for the old one within the new widgets API. However, the widgets API is still behind a feature flag, and that is gonna remain like that for a while. So we need to also expose the new components _outside_ of the widgets API, integrated into the (messy) legacy infrastructure, so that it is also displayed in a production build (`npm run build`). We're doing something similar with the PageSpeed Insights widget, where the reused component `DashboardPageSpeed` is wrapped in another component which is then added to the legacy infrastructure via `addFilter`.

**Comment by felixarntz** [comment]
IB ✅

**Comment by felixarntz** [comment]
@benbowler Just wanted to mention: While this is technically blocked by #2422 which is still being worked on, I believe you only need to touch the outer `DashboardAllTrafficWidgetV2` component for this issue, so probably you can already start the PR (just being aware that the area of the widget that will contain the graph is currently still empty).

**Comment by felixarntz** [comment]
@benbowler @aaemnnosttv There are a couple UI pieces to still address before we launch the new widget in production. I'll open issues for those pieces, and this one is now for 1.25.0. Let's still address the final quirks here, afterwards we can have it wait in CR until the widget has been approved for launch.

**Comment by felixarntz** [comment]
@benbowler Opened the necessary follow-up issues to get the widget ready for production. I've added them as blockers here. They're technically only merge blockers, the code for this issue shouldn't be affected.

**Comment by benbowler** [comment]
Thanks @felixarntz, I'll track those blockers and come back to this ticket when ready to get it back into Code Review.

**Comment by benbowler** [comment]
There is a remaining issue to get this pushed, I've spent some time investigating but I haven't been able to resolve it, @eugene-manuilov / @aaemnnosttv can you take a look at [this comment](https://github.com/google/site-kit-wp/pull/2617#issuecomment-764733855)

I've tried adding keys to the Tab bar and chart elements as well as GoogleCharts as well as using useEffect to force re-rendering without luck.

**Comment by benbowler** [comment]
Issue [fixed](https://github.com/google/site-kit-wp/pull/2617#issuecomment-765481882) and ready for review.

**Comment by benbowler** [comment]
Update use of `useInstanceID`.

**Comment by benbowler** [comment]
Added chartID to the onReady useCallback dependencies list.

**Comment by felixarntz** [comment]
**QA** ⚠️ 

Forwarding test feedback from @marrrmarrr here: When activating the PageSpeed Insights module, the blue link to jump to the PSI widget on the dashboard now goes to All Traffic instead.

@benbowler @eugene-manuilov @aaemnnosttv Could one of you take a look at this?

**Comment by aaemnnosttv** [comment]
@felixarntz simple fix – PR ready for review targeting `master`.

**Comment by cole10up** [comment]
### QA ❌

**Tested:**
On my personal site, I ran the following test using **Chrome:**

1. Reset, deactivated and deleted the prior version installed
2. Installed, activated and setup the zip above
3. Setup the Analytics module and navigated to the dashboard

Notice two graphs, one empty and one with data:
![Site Kit by Google Dashboard ‹ SupremeQA — WordPress](https://user-images.githubusercontent.com/57115907/106226468-247e9500-619c-11eb-97fb-11fab41a00c7.gif)

No console errors are present:
![image](https://user-images.githubusercontent.com/57115907/106226530-4710ae00-619c-11eb-8e95-e55299f91c49.png)

Added @felixarntz and @aaemnnosttv to the site for analysis.

**Comment by wpdarren** [comment]
@cole10up did you clear your browser cache? I was just able to recreate this with the zip, but when I cleared cache, it displayed as expected.  c.c. @aaemnnosttv @felixarntz

**Comment by aaemnnosttv** [comment]
@cole10up – I tested with the latest zip in Chrome, Firefox, and Safari and wasn't able to reproduce what you're seeing here.

I also tried logging in on your site to see if I could see it there but I don't have access to the configured Analytics profile so the widget doesn't show at all for me.

Sending this back to QA for another pass as it seems most likely like a cache issue. Can you take another look @wpdarren ?

**Comment by wpdarren** [comment]
### QA Update:

**Verified:**

- On the dashboard, the All Traffic Widget must be updated to the v2.
- All Traffic heading must be changed to `Your Traffic at a Glance`.

For reference I used the `master` branch to test this. 
Cleared cache.  Disconnected and reset Site Kit. Then Connected Google account and Analytics. 

![image](https://user-images.githubusercontent.com/73545194/106275693-35e5a100-6271-11eb-8064-b80185ad3ac0.png)

@aaemnnosttv sorry, but please could you clarify what this means in the QAB. 

- All Traffic Widget story under Dashboard and Page Dashboard must be replaced with the v2 of the widget.
- All Traffic Widget V2 story under Dashboard must be removed.

I can see V2 is appearing in the dashboard as per above.

**Comment by aaemnnosttv** [comment]
@wpdarren – this is referring to removing stories in storybook which render the old version of the all traffic widget (which should now be completely removed). For example: [this one](https://google.github.io/site-kit-wp/storybook/develop/?path=/story/dashboard--all-traffic).

**Comment by wpdarren** [comment]
### QA Update: Passed ✅ 

**Verified:**

- On the dashboard, the All Traffic Widget is updated to v2.
- All Traffic heading is changed to `Your Traffic at a Glance`.
- The old widget in storybook is removed from the dashboard 

For reference I used the `master` branch to test this. 
Cleared cache.  Disconnected and reset Site Kit. Then Connected Google account and Analytics. 

![image](https://user-images.githubusercontent.com/73545194/106275693-35e5a100-6271-11eb-8064-b80185ad3ac0.png)

**Comment by aaemnnosttv** [comment]
@felixarntz this wasn't in the ACs (or IB) but there is one remaining story that has the old all traffic widget in it:
https://github.com/google/site-kit-wp/blob/9da5402e860b9333022b2082dfd74cbec0709f22/stories/dashboard.stories.js#L56
This is essentially just a wrapper for `LegacyDashboardAcquisitionPieChart` and `LegacyAnalyticsAllTrafficDashboardWidgetTopAcquisitionSources` though.

Should we address it as part of this or in a follow-up issue?

**Comment by felixarntz** [comment]
@aaemnnosttv No huge preference, if this is something we can quickly do, I think it's least work to open a follow-up PR against this issue.
